src/v/cluster_link/replication/partition_replicator.cc

```diff
@@ -199,6 +199,7 @@ ss::future<> partition_replicator::replicate_and_wait(
             .finally(
               [inflight = std::move(inflight), data = std::move(data)]() {});
       });
+    co_await _sink->maybe_sync_pid();
```

Comment:
We replicate first (async) and then we sync pid (sync)? Is the method name misleading?

Shouldn't it be a bit the other way around? First ensure that there won't be a PID collision, then replicate.

Otherwise, the collision may still happen the way I read it i.e. if sync pid throws/dies or etc.

Reply:
Yeah, it’s a best-effort update. If we make it pre-replication and id_allocator is leaderless, replication stops completely. Not sure if this pid sync thing is a hard requirement for the first version.. maybe something we can improve later.

Reply:
Ok cool, just confirming. I think either approach risks a bit of data loss somewhere - like if we stall replication here, then the target falls a bit behind the head of the source log.

I guess we could build something more sophisticated post-GA, but absent that, I think prioritizing source -> target replication makes sense.